src/isa/ebpf/canonicalize.cc

```diff
@@ -1424,20 +1424,39 @@ void safety_chk_insn(inst& insn, inst_static_state& iss) {
     safety_chk_insn_mem_access(insn, iss);
   }
 }
-
-void static_safety_check_pgm(inst * program, int len) {
+/*
+* returns the number of unsafe instructions in the program
+*
+*/
+int static_safety_check_pgm(inst * program, int len) {
+  // pss: program static state
+  // going through instructions and inferring the states.
   prog_static_state pss;
   pss.static_state.resize(len + 1);
   pss.g.gen_graph(program, len);
   topo_sort_for_graph(pss.dag, pss.g);
   type_const_inference_pgm(pss, program, len);
   min_pkt_sz_inference_pgm(pss, program, len);
+  // utilizing static states to check safety of each instruction.
+  int num_of_unsafe_ins = 0;
   for (int i = 0; i < len; i++) {
-    safety_chk_insn(program[i], pss.static_state[i]);
+    try{
+      //throws a string error message if instruction is unsafe
+      safety_chk_insn(program[i], pss.static_state[i]);
+    }
+    catch(string err_msg){
+      num_of_unsafe_ins++;
+      cout << err_msg << endl;
```

Comment:
This function will be called for each synthesis in the searching procedure. Printing the error message for each unsafe program seems verbose (it's quite often that the sample program is unsafe). How about printing it when the logger level is set as LOGGER_DEBUG?

Reply:
Also, I am thinking about whether it would be better to throw the error if there is an error being captured? Since it helps keep the same logic as the previous code which throws errors. and it will help run the code faster: once an unsafe instruction is detected, the compiler does not need to check the remaining instructions.